### ISSUE TITLE
How to use Autopool for SED

### ISSUE THREAD

**Issue by yuvaramsingh94** [marl/autopool#10]
Hi @justinsalamon ,
    i am currently trying to implement the MIL approach for SED based on your adaptive pooling paper. [https://arxiv.org/pdf/1804.10070.pdf](url). currently my implementation is not giving the intended results . i like to get your opinion on my implementation and correct me if i have missed any.

1.split the single track into 4 instance of 5sec each
2, converted them into melspec and passed to feature extractor made of conv layer followed by fully connected layer .
3, at the end of the previous step, the model will give sigmoid output for 4instance X n_classes.
4, this is passed on to my implememtation of Autopooling layer and will get a single  vector containing n_classes weighted sigmoid values.
5, i am using BCEWithLogitsLoss in pytorch which is equivalent to BinaryCrossentropy with sigmoid layer in keras

pls share your thoughts on this.
thanks 
yuvaram

**Comment by justinsalamon** [comment]
Sounds about right (we used much shorter instances though). That said - I wouldn't develop an implementation against an actual problem. Instead I would test it on toy data to confirm the implementation is correct and compare it to the Keras implementation on the same data to confirm.

If the implementation works correctly and your model isn't performing as you expect, it's a research problem, not a code problem.

**Comment by yuvaramsingh94** [comment]
Thanks. i have got some good results after making some changes to the softmax implementation. this gave me a big boost compared to the older model where i was not using any weekly supervised learning method. so i hope i have done something right with the implementation.

**Comment by justinsalamon** [comment]
Glad it's working for you. It's probably still a good idea to write tests and use toy data to verify the implementation is correct -  this way you can go from hoping to knowing :)
